server/tests/api/document.spec.js

```diff
@@ -0,0 +1,253 @@
+import chai from 'chai';
+import expect from 'expect';
+import chaiHttp from 'chai-http';
+import fakeDocuments from '../__mock__/fakeDocuments';
+import fakeUsers from '../__mock__/fakeUsers';
+import server from '../../server';
+
+const { superAdminLoginDetail, adminLoginDetail, anotherRegularUserDetail } = fakeUsers;
+const { privateDocument, roleDocument, invalidFieldDocument } = fakeDocuments;
+let superAdminToken, adminToken, regularUserToken;
+
+chai.use(chaiHttp);
+
+describe('Documents', () => {
+  before((done) => {
+    chai.request(server)
+      .post('/api/v1/users/login')
+      .send(superAdminLoginDetail)
+      .end((err, res) => {
+        superAdminToken = res.body.tokenLogin;
+        done();
+      });
+  });
+  before((done) => {
+    chai.request(server)
+      .post('/api/v1/users/login')
+      .send(adminLoginDetail)
+      .end((err, res) => {
+        adminToken = res.body.tokenLogin;
+        done();
+      });
+  });
+
+  before((done) => {
+    chai.request(server)
+      .post('/api/v1/users/login')
+      .send(anotherRegularUserDetail)
+      .end((err, res) => {
+        regularUserToken = res.body.tokenLogin;
+        done();
+      });
+  });
+
+  describe('POST /api/v1/documents', () => {
+    it('should create a new document with a status code 200', (done) => {
+      chai.request(server)
+      .post('/api/v1/documents')
+      .set({ authorization: regularUserToken })
+      .send(privateDocument)
+      .end((err, res) => {
+        expect(res.status).toBe(200);
```

Comment:
Just checking for the status codes is usually not enough to assert that the controller functions the way it should, try to check that the response data/body is what you expect. Fix this everywhere else it applies.